### ISSUE TITLE
Use AbstractMCMC interface

### ISSUE THREAD

**Issue by sethaxen** [jpsamaroo/PRS.jl#9]
Instead of implementing MCMC itself, this package could use the AbstractMCMC.jl interface to do sampling. This would enable various MCMC flavors in different packages to be used for the sampling, e.g.
- AdvancHMC.jl: HMC/NUTS
- AdvancedMH.jl: Metropolis-Hastings
- AdvancedPS.jl: particle sampling

AdvancedMCMC.jl also has functionality for sampling multiple chains in parallel using multiple threads or cores.

**Comment by jpsamaroo** [comment]
Agreed! Unfortunately neither @fauconab nor I are very familiar with MCMC; we ported this package mostly as-is from Python because we needed the core algorithm to run faster for our research. We would appreciate any help with figuring out how to generalize the MCMC so that Julia-native approaches could be tried out instead!
